airflow/exceptions.py

```diff
@@ -225,3 +226,30 @@ def __str__(self):
 
 class ConnectionNotUnique(AirflowException):
     """Raise when multiple values are found for the same conn_id"""
+
+
+class TaskDeferred(BaseException):
+    """
+    Special exception raised to signal that the operator it was raised from
+    wishes to defer until a trigger fires.
+    """
+
+    def __init__(
+        self,
+        *,
+        trigger,
+        method_name: str,
+        kwargs: Optional[Dict[str, Any]] = None,
+        timeout: Optional[datetime.timedelta] = None,
+    ):
+        super().__init__()
+        self.trigger = trigger
+        self.method_name = method_name
+        self.kwargs = kwargs
+        self.timeout = timeout
+        # Check timeout type at runtime
+        if self.timeout is not None and not hasattr(self.timeout, "total_seconds"):
+            raise ValueError("Timeout value must be a timedelta")
```

Comment:
Curious why not `isinstance(self.timeout, datetime.timedelta)`?

Reply:
I still prefer duck-typing to instance checking because I've been doing Python for Too Long. Happy to change it if you'd prefer!

Reply:
Should be fine -- was just curious